### PR TITLE
PartDesignNext rebased on master

### PR DESCRIPTION
**Pull request by ickby**
As discussed in http://forum.freecadweb.org/viewtopic.php?f=10&t=15207 this is a pull request for the PartDesign Next branch. There may be adoptions needed before merging, lets use this Pull Request for technical discussions if the need arises.

### CODE REVIEW

File: .kdev4/.kdev4/FCMasterMerge.kdev4 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Buildset]
```
Comment (f3nix) [inline .kdev4/.kdev4/FCMasterMerge.kdev4]: I think KDevelop files shouldn't be merged.

Comment (ickby) [inline .kdev4/.kdev4/FCMasterMerge.kdev4]: Yes true, no idea how this got added.

**Comment by bblacey** [comment]
@ickby, @wwmayer recently committed a [new travis build config](https://github.com/FreeCAD/FreeCAD/commit/0eaf34ce18bfb9015afae2bf43a1892dee0df64d) to master that moves the linux builds from precise with gcc to trusty with clang and C++11 hence if you rebase your pull request with master, the pull request will be rebuilt with the new Linux configuration.  As @sasobadovinac demonstrated, you will also want to apply [@wwmayer's patch in saso's fork](https://github.com/sasobadovinac/FreeCAD/commit/a788528f3e93ae7e4b4e54ac8e8d409715888159.patch) - I have confirmed these changes on [a forked build of PR-135](https://travis-ci.org/bblacey/FreeCAD-MacOS-CI/jobs/122175756)

**Comment by ickby** [comment]
Ok all comments have been incorporated. Merging should be ok now.

**Pull request merged**